modules/core/kotlin/com/github/timaxinc/kcipt/classloader/WhitelistClassloader.kt

```diff
@@ -0,0 +1,139 @@
+package com.github.timaxinc.kcipt.classloader
+
+import com.github.timaxinc.kcipt.util.io.Block
+import com.github.timaxinc.kcipt.util.io.startsWithMember
+import java.net.URL
+import java.util.*
+
+/**
+ * WhitelistClassloader is a ClassLoader with a predefined Whitelist of Classes and packages that it will allow to
+ * load resources and Classes from. All other requests will be blocked
+ *
+ * @property softMode
+ *          If activated, softMode will prevent an Exception to be thrown, if the Class is on the blacklist
+ * @property whitelist
+ *          A List of Strings containing the names of Classes packages and resources that are to be whitelisted. Only
+ *          those Classes/resources will be permitted to load, while all other requests will be blocked.
+ * @constructor
+ *          Instantiates a new WhitelistClassloader.
+ *
+ * @param parent
+ *          the parent ClassLoader
+ */
+class WhitelistClassloader(
```

Comment:
Should only implement a method that is used to decide if a class, package or resource is allowed to access.

Reply:
Please elaborate further... How would replacing the current method with the method suggested by you ensure the planned functionality? Furthermore I see no way to determine if a Class or a resource (or all for that matter) are requested.